lib/smart_todo/slack_client.rb

```diff
@@ -55,7 +55,10 @@ def lookup_user_by_email(email)
     #
     # @see https://api.slack.com/methods/chat.postMessage
     def post_message(channel, text)
-      request(:post, "/api/chat.postMessage", JSON.dump(channel: channel, text: text))
+      headers = default_headers
+      request = Net::HTTP::Post.new("/api/chat.postMessage", headers)
+      request.body = JSON.dump(channel: channel, text: text)
+      dispatch(request)
```

Comment:
This is interesting! I don't know `net/http` well and I would have assumed it would mutate the request object. I like this though.